test/test_expr.c

```diff
@@ -64,14 +64,17 @@ int lookup(void *data, char *str, char **end, hts_expr_val_t *res) {
         kputs("", ks_clear(&res->s));
     } else if (strncmp(str, "null-but-true", 13) == 0) {
         *end = str+13;
+        hts_expr_val_undef(res);
```

Comment:
Maybe this should be done using a `nan-but-true`, leaving the `null-but-true` test as-is.

Note that `nan-but-true` could occur, if a floating-point tag exists with a `nan` in it.

Reply:
I switched to using nan at your own request...

I'm not inclined to switch back again now.  I don't think we really need to worry too much about real nans as it's such a corner case we can simply document it as nan being considered to be false by default (which it is in hts_expr.h).

I thought I'd already updated the expression syntax documentation too, but apparently not.  (It's rather annoying that most of htslib's documentation is in samtools, but that's a different issue.)